client/windows/CHeroOverview.cpp

```diff
@@ -0,0 +1,256 @@
+/*
+ * CHeroOverview.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+#include "CHeroOverview.h"
+
+#include "../CGameInfo.h"
+#include "../gui/CGuiHandler.h"
+#include "../render/Canvas.h"
+#include "../render/Colors.h"
+#include "../render/Graphics.h"
+#include "../render/IImage.h"
+#include "../renderSDL/RenderHandler.h"
+#include "../widgets/CComponent.h"
+#include "../widgets/Images.h"
+#include "../widgets/TextControls.h"
+
+#include "../../lib/CGeneralTextHandler.h"
+#include "../../lib/CCreatureHandler.h"
+#include "../../lib/CHeroHandler.h"
+#include "../../lib/CSkillHandler.h"
+#include "../../lib/spells/CSpellHandler.h"
+
+CHeroOverview::CHeroOverview(const HeroTypeID & h)
+	: CWindowObject(BORDERED | RCLICK_POPUP), hero { h }
+{
+	OBJ_CONSTRUCTION_CAPTURING_ALL_NO_DISPOSE;
+
+    heroIndex = hero.getNum();
+
+	pos = Rect(0, 0, 600, 485);
+
+	genBackground();
+    genControls();
+
+	center();
+}
+
+void CHeroOverview::genBackground()
```

Comment:
Better option is to use showAll method and:
1) show background texture
2) draw this overlay
3) call CIntObject::showAll() to draw all controls

Reply:
I did not understand it exactly. :(